tests/unit/test_http_session.py

```diff
@@ -367,6 +367,26 @@ def test_proxy_ssl_context_uses_check_hostname(self):
             last_call = self.proxy_manager_fun.call_args[-1]
             self.assertIs(last_call['ssl_context'].check_hostname, True)
 
+    def test_proxy_ssl_context_does_not_use_check_hostname_if_ip_address(self):
+        cert = ('/some/cert', '/some/key')
+        proxies_config = {'proxy_client_cert': "path/to/cert"}
+        urls = ['https://1.2.3.4:5678',
+                'https://4.6.0.0',
+                'https://[FE80::8939:7684:D84b:a5A4%251]:1234',
```

Comment:
Let's make sure we're doing IP's without a zone identifier or port too.

i.e.
* https://[FE80::8939:7684:D84b:a5A4]
* https://[FE80::8939:7684:D84b:a5A4]:1234